hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java

```diff
@@ -71,11 +72,36 @@ private boolean isSorted(List<Cell> list) {
     }
 
     public void mutate(List<Cell> rawList, List<TxnView> txns, List<Cell> results) throws IOException {
-        assert dataToReturn.isEmpty();
-        assert results.isEmpty();
-        assert maxTombstone == null;
-        assert !LOG.isDebugEnabled() || isSorted(rawList) : "CompactionStateMutate: rawList not sorted";
-        assert rawList.size() == txns.size();
+        final boolean dataToReturnIsEmpty = dataToReturn.isEmpty();
+        final boolean resultsIsEmpty = results.isEmpty();
+        final boolean maxTombstoneIsNull = maxTombstone == null;
+        final boolean rawListAndTxnListSameSize = rawList.size() == txns.size();
+        final boolean debugSortCheck = !LOG.isDebugEnabled() || isSorted(rawList);
+        if (!debugSortCheck) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  rawList is not sorted.");
+        }
+        if (!dataToReturnIsEmpty) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  dataToReturn is not properly initialized.");
+        }
+        if (!resultsIsEmpty) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  results list not properly initialized.");
+        }
+        if (!maxTombstoneIsNull) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  maxTombstone not properly initialized to null.");
+        }
+        if (!rawListAndTxnListSameSize) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  rawList and txn list not the same length.");
+        }
+        assert dataToReturnIsEmpty;
+        assert resultsIsEmpty;
+        assert maxTombstoneIsNull;
+        assert debugSortCheck : "CompactionStateMutate: rawList not sorted";
+        assert rawListAndTxnListSameSize;
```

Comment:
Can you move all these checks to a separate method to bring into focus the logic of the method?

Reply:
Done.